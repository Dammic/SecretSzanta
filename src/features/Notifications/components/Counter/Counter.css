
.counter {
    color: black;
    font-weight: bold;
}
