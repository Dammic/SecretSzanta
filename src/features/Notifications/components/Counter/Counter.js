import React from 'react'
import PropTypes from 'prop-types'
import styles from './Counter.css'

export class Counter extends React.PureComponent {
    static displayName = 'Counter'
    static propTypes = {
        start: PropTypes.number,
    }

    state = { counter: this.props.start }

    componentDidMount() {
        this.interval = setInterval(this.countDown, 1000)
    }

    componentWillUnmount() {
        clearInterval(this.interval)
    }

    countDown = () => {
        this.setState((prevState) => {
            const newCounter = prevState.counter - 1

            if (newCounter <= 0) {
                clearInterval(this.interval)
                return { counter: null }
            }

            return { counter: newCounter }
        })
    }

    render() {
        return (
            <span className={styles.counter}>
                {this.state.counter}sec
            </span>
        )
    }
}
