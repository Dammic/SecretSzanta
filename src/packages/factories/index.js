export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
