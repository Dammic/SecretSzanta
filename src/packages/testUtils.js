import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
