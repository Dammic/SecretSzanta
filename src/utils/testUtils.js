import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
