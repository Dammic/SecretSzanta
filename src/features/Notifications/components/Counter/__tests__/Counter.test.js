
import React from 'react'
import { shallow } from 'enzyme'
import { expectShallowMatchingSnapshot, asyncWait } from 'packages/testUtils'
import { Counter } from '../Counter'

jest.useFakeTimers()

describe('<Counter />', () => {
    it('matches snapshot at beginning of countdown', () => {
        expectShallowMatchingSnapshot(<Counter start={2} />)
    })

    it('counts down to 0', async () => {
        const component = shallow(<Counter start={5} />)
        jest.runAllTimers()
        await asyncWait()
        expect(clearInterval).toHaveBeenCalledTimes(1)
        expect(component.state('counter')).toEqual(null)
    })
})
