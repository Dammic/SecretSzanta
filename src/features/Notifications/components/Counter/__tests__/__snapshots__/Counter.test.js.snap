// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> matches snapshot at beginning of countdown 1`] = `
<span
  className="counter"
>
  2
  sec
</span>
`;
